subprojects/core/src/main/java/org/gradle/api/internal/provider/DefaultProviderFactory.java

```diff
@@ -69,9 +67,9 @@ public DefaultProviderFactory(FileOperations fileOperations) {
             ((PropertyState<Double>) propertyState).set(Double.valueOf(0));
         } else if (clazz == Character.class) {
             ((PropertyState<Character>) propertyState).set(new Character('\0'));
-        } else if (clazz == FileCollection.class || clazz == ConfigurableFileCollection.class) {
+        } else if (clazz == ConfigurableFileCollection.class) {
```

Comment:
yes, except for this:

> It also means that FileCollection types cannot be used for mapping extension properties to task properties until the next story is implemented

FileCollection is already a provider, and ConfigurableFileCollection is already a property state:

```
class MyTask {
    private final ConfigurableFileCollection someFiles = project.files()

    FileCollection getSomeFiles() { return someFiles }

    void setSomeFiles(Collection<File> files) { someFiles.setFrom(files) }
    void setSomeFiles(FileCollection files) { someFiles.setFrom(files) }
}

myTask.someFiles = project.files { someExtension.someFiles }
someExtension.someFiles = file("a")
assert myTask.someFiles.files == [file("a")]
```